client/src/app/site/assignments/components/assignment-poll/assignment-poll.component.ts

```diff
@@ -0,0 +1,207 @@
+     */
+    public constructor(
+        public pollService: AssignmentPollService,
+        private operator: OperatorService,
+        private assignmentRepo: AssignmentRepositoryService,
+        public translate: TranslateService,
+        public dialog: MatDialog,
+        private promptService: PromptService
+    ) {}
+
+    /**
+     * TODO
+     */
+    public ngOnInit(): void {}
+
+    /**
+     * Handler for the 'delete poll' button
+     *
+     * TODO: Some confirmation (advanced logic (e.g. not deleting published?))
+     */
+    public async onDeletePoll(): Promise<void> {
+        const title = this.translate.instant('Are you sure you want to delete this poll?');
+        if (await this.promptService.open(title, null)) {
+            await this.assignmentRepo.deletePoll(this.poll.id);
+        }
+        // TODO error handling
+    }
+
+    /**
+     * TODO Print the ballots for this poll.
+     */
+    public printBallot(poll: Poll): void {
+        this.promptService.open('TODO', 'TODO');
+        // TODO Print ballot not implemented
+    }
+
+    /**
+     * Fetches the name for a candidate from the assignment
+     *
+     * @param option Any poll option
+     * @returns the full_name for the candidate
+     */
+    public getCandidateName(option: PollOption): string {
+        const user = this.assignment.candidates.find(c => c.id === option.candidate_id);
+        return user ? user.full_name : '';
+        // TODO this.assignment.candidates may not contain every candidates' name (if deleted later)
+        // so we should rather use this.userRepo.getViewModel(option.id).full_name
+        // TODO is this name always available?
+        // TODO error handling
+    }
+
+    /**
+     * Determines whether the candidate has reached the majority needed to pass the quorum
+     *
+     * @param option
+     * @returns true if the quorum is successfully met
+     * TODO check
+     */
+    public quorumReached(option: PollOption): boolean {
+        const amount = option.votes.find(v => v.value === 'Yes').weight;
+        return amount > this.pollService.yesQuorum(this.poll, option);
+    }
+
+    /**
+     * Opens the {@link AssignmentPollDialogComponent} dialog and then updates the votes, if the dialog
+     * closes successfully (validation is done there)
+     *
+     */
+    public enterVotes(): void {
+        // TODO deep copy of this.poll (JSON parse is ugly workaround)
+        // or sending just copy of the options
+        const data = {
+            poll: JSON.parse(JSON.stringify(this.poll)),
+            users: this.assignment.candidates // used to get the names of the users
+        };
+        const dialogRef = this.dialog.open(AssignmentPollDialogComponent, {
+            data: data,
+            maxHeight: '90vh',
+            minWidth: '300px',
+            maxWidth: '80vh',
+            disableClose: true
+        });
+        dialogRef.afterClosed().subscribe(result => {
```

Comment:
maybe make this async: `const result = await dialogRef.afterClosed().toPromise();`

Reply:
I'm not sure this will really work (I used the dialogRef as it was thrown to me in the angular docs). If you're sure, I can change it, else I'll look into it later

Reply:
okay, this works. But currently we have the other variant about a dozen times all over the place, so the change would introduce some inconsistency